### PR TITLE
Feature/ocsp

### PR DESCRIPTION
**Pull request by gianluca-isgro**
OCSP check integration.

### CODE REVIEW

**Comment by IgorBalog-Eng** [comment]
What about adding short description in CHANGELOG file, regarding new functionality?

**Review by IgorBalog-Eng — COMMENTED** [review]
This change is when consumer sends request to provider (sender -> receiver)
What about when provider sends response back to consumer? (receiver -> sender)

File: src/main/java/it/eng/idsa/businesslogic/util/OCSPValidation.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+			logger.error("File " + tmpdir + host + ".pem is empty.");
+			
+			hostCertificate.delete();
+			
+			return OCSP_STATUS.unknown;
+		}
+		
+		/* get OCSP URI if exist */
+		cmdList = new ArrayList<>();
+		cmdList.add("/bin/sh");
```
[the remark this answers is not in the document]
Comment (gianluca-isgro) [reply src/main/java/it/eng/idsa/businesslogic/util/OCSPValidation.java:111]: What do you suggest? Could do something like this

if(System.getProperty("os.name").toLowerCase().contains("win")) {
			logger.info("Running on windows");
			startCmdList.add("cmd.exe");
			startCmdList.add("/c");
		} else {
			startCmdList.add("/bin/sh");
			startCmdList.add("-c");
		}

but to work under Windows you should have OpenSSL installed and entered in the path. I'm also not sure if the parameters are the same and can be passed in the same way.

[the remark this answers is not in the document]
Comment (IgorBalog-Eng) [reply src/main/java/it/eng/idsa/businesslogic/util/OCSPValidation.java:111]: Could be used as solution for OS decision.

[the remark this answers is not in the document]
Comment (gianluca-isgro) [reply src/main/java/it/eng/idsa/businesslogic/util/OCSPValidation.java:111]: Okay but it does not seem to me a response to my comment. 
If I implement it, it should be verified to work in a Windows environment.

[the remark this answers is not in the document]
Comment (IgorBalog-Eng) [reply src/main/java/it/eng/idsa/businesslogic/util/OCSPValidation.java:111]: I can volunteer to test this in Windows, but I know that 'sed' command is not present in windows, so you will have to come up with some other solution for that command

**Comment by gianluca-isgro** [comment]
> This change is when consumer sends request to provider (sender -> receiver) What about when provider sends response back to consumer? (receiver -> sender)

I tried to figure out where to put the OCSP check. I had thought of the ReceiverSendDataToBusinessLogicProcessor class but I can't find the hostname of the sender. I don't think it's there there.

**Comment by IgorBalog-Eng** [comment]
> > This change is when consumer sends request to provider (sender -> receiver) What about when provider sends response back to consumer? (receiver -> sender)
> 
> I tried to figure out where to put the OCSP check. I had thought of the ReceiverSendDataToBusinessLogicProcessor class but I can't find the hostname of the sender. I don't think it's there there.

For now, let's postpone this receiver certificate check

**Review by davidjovanovic — APPROVED** [review]

**Pull request merged**